src/components/fabSpeedDial/fabSpeedDial.js

```diff
@@ -0,0 +1,138 @@
+(function() {
+  'use strict';
+
+  angular
+    .module('material.components.fabSpeedDial', [])
+    .directive('mdFabSpeedDial', MdFabSpeedDialDirective);
+
+  /**
+   * @ngdoc directive
+   * @name mdFabSpeedDial
+   * @module material.components.fabSpeedDial
+   *
+   * @restrict E
+   *
+   * @description
+   * The `<md-fab-speed-dial>` directive is used present a series of popup elements (usually `<md-button>`s) for
+   * quick access to common actions.
+   *
+   * @usage
+   * <hljs lang="html">
+   * <md-fab-speed-dial direction="up">
+   *   <md-fab-trigger>
+   *     <md-button><md-icon icon="/img/icons/plus.svg"></md-icon></md-button>
+   *   </md-fab-trigger>
+   *
+   *   <md-fab-actions>
+   *     <md-button><md-icon icon="/img/icons/user.svg"></md-icon></md-button>
+   *     <md-button><md-icon icon="/img/icons/group.svg"></md-icon></md-button>
+   *   </md-fab-actions>
+   * </md-fab-speed-dial>
+   * </hljs>
+   *
+   * @param {string=} md-direction From which direction you would like the speed dial to appear relative to the
+   * trigger element.
+   * @param {expression=} md-open Programmatically control whether or not the speed-dial is visible.
+   */
+  function MdFabSpeedDialDirective($timeout) {
```

Comment:
Can you refactor the code such that the "directive" is just a simple configuration object, where the `compile`, `link`, and `controller` properties are set as references to names functions?